### DISCUSSION TITLE
Issue: Table works slow and is laggy with bigger DB

### DISCUSSION THREAD

**Discussion by Gentleman03** [RafaelGB/obsidian-db-folder#277] [General]
**Describe the bug**
Bigger db (in my case 121 entries (rows) atm) makes filters and adding a new file (row) slow and whole table becomes a bit laggy.
For example i have table with books and I want to display books that belong to certain author, then when I remove this filter, I have to wait at least couple of seconds to get view of whole table again (i guess generating table takes all this time). When I edit content of any column, after pressing enter whole table freezes for couple of seconds. It makes working with table very annoying.

Another issue is that writing in "filename of new row" field is VERY laggy. Of course with bigger DB, there is no issue with small DB.

**To Reproduce**
Just use DB with 100+ files. I can share my DB but I would like to keep it private so I would need some way to share it other way than just upload somewhere and share link here for everyone.

_Steps to reproduce the behavior:_
1. Go to 'search field' and write something to filter records in table
2. Delete filter
3. Table freezes

1. Go to 'filename of new row' field and just write anything
2. Table freezes

1. Go to any column, edit it's content
2. Press enter
3. Table freezes

**Expected behavior**
Table working without freezing.

**Desktop:**
 - OS: Windows 10 Pro
 - Latest version of Obsidian and DB Folder but it always was like that, no matter which version of Obsidian or plugin I was using.

It's such a great plugin but I struggle so hard working with it because of those freezes :(. I hope there is some way to get rid of those freezes.

**Comment by RafaelGB** [comment]
with 2.1.0 the performance was improved, but depends of your cpu, the number of columns and off the number of rows.

In theory the used library supports a max of 100k rows. We are always working on improving it

I will move this issue to discussion being a request for continuous improvement

**Comment by RafaelGB** [comment]
as a performance ddbb had +300 and should work

https://user-images.githubusercontent.com/11924043/185806305-852d7ec3-4cc5-4d5f-a1b0-6892be67f52e.mov

**Comment by Gentleman03** [comment]
> with 2.1.0 the performance was improved, but depends of your cpu, the number of columns and off the number of rows.
> 
> In theory the used library supports a max of 100k rows. We are always working on improving it
> 
> I will move this issue to discussion being a request for continuous improvement

Well, my CPU is pretty old but it's still 4x2,83 GHz so it should be fine? My CPU usage doesn't reach 100% (not even with highest peaks) when working with table so maybe it's not CPU? I tried to move my DB folder to SSD but havent noticed any significant improvement.

>as a performance ddbb had +300 and should work

It is working but it's laggy and slow. Actually editing columns content is the fastest operation, just try to write something in "filename for new row" field and see how long it will take for the text to appear in the field.

Maybe it would be possible to create some kind of table cache to avoid reloading whole table when it's not necessery?

**Reply by RafaelGB** [reply]
actually the plugin is using Zustand as the state manager, was the improvement of the 2.1.0 instead of redux where all was reloaded.

I am noob with that technology and surely could be improved a lot but we are in the right direction

If we find a way to separate every row with its own manager I think the performance could be astounding. The problem with that are some actions like remove columns or change the type of a column

**Reply by Gentleman03** [reply]
So it can be done, great to hear that. I hope some day you'll be able to make it happen. 
Good luck and thanks for amazing plugin!

**Reply by RafaelGB** [reply]
try with 2.2.2 . The performance was improved a bit

**Reply by haemi** [reply]
my folder currently has about 4k files and is very, very slow; is there hope for even more performance improvements?

**Reply by RafaelGB** [reply]
The changer will be https://github.com/RafaelGB/obsidian-db-folder/issues/116

I will work on it after finish the autocomplete issue

**Reply by RafaelGB** [reply]
I just up a demo to that issue so u can see the pagination working
